src/huggingface_hub/hf_api.py

```diff
@@ -4070,6 +4070,69 @@ def restart_space(self, repo_id: str, *, token: Optional[str] = None) -> SpaceRu
         hf_raise_for_status(r)
         return SpaceRuntime(r.json())
 
+    @validate_hf_hub_args
+    def duplicate_space(
+        self,
+        src_repo_id: str,
+        *,
+        dest_repo_id: Optional[str] = None,
+        private: Optional[bool] = None,
+        token: Optional[str] = None,
+    ) -> str:
```

Comment:
Nice PR @nateraw , thanks for implementing it. I haven't reviewed properly yet (have to double-check how the endpoint works) but I think we should rename the inputs as `from_id` and `to_id`. Not that I'm strongly opinionated about it but it would be the same syntax as `HfApi.move_repo(...)`.

```suggestion
    def duplicate_space(
        self,
        from_id: str,
        to_id: Optional[str] = None,
        *, 
        private: Optional[bool] = None,
        token: Optional[str] = None,
    ) -> str:
```

Reply:
ah perfect. yea that was one thing I wasn't too sure about and wanted your opinion on. Thanks!